relay-metrics/src/aggregation.rs

```diff
@@ -1200,14 +1269,19 @@ impl Aggregator {
 
         let key = Self::validate_bucket_key(key, &self.config)?;
 
+        let added_cost;
         match self.buckets.entry(key) {
             Entry::Occupied(mut entry) => {
                 relay_statsd::metric!(
                     counter(MetricCounters::MergeHit) += 1,
                     metric_type = entry.key().metric_type.as_str(),
                     metric_name = &entry.key().metric_name
                 );
-                value.merge_into(&mut entry.get_mut().value)?;
+                let bucket_value = &mut entry.get_mut().value;
+                let cost_before = bucket_value.cost();
+                value.merge_into(bucket_value)?;
+                let cost_after = bucket_value.cost();
+                added_cost = cost_after.saturating_sub(cost_before);
```

Comment:
We could probably optimize this by making `merge_into` return the actual cost delta, but I decided against it for the sake of simplicity.

Reply:
Wouldn't it be simpler to just add the cost of the single value here and have `merge_into` return whether or not something was added?

Reply:
Yeah that could work.